clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -1,30 +1,54 @@
 package org.broadinstitute.clio.client.commands
 
-import akka.http.scaladsl.model.HttpResponse
-import enumeratum._
-import org.broadinstitute.clio.client.parser.BaseArgs
-import org.broadinstitute.clio.client.webclient.ClioWebClient
-
-import scala.collection.immutable.IndexedSeq
-import scala.concurrent.{ExecutionContext, Future}
-
-sealed trait CommandType extends EnumEntry
-
-object Commands extends Enum[CommandType] {
-  override val values: IndexedSeq[CommandType] = findValues
+import java.time.OffsetDateTime
+
+import caseapp.Recurse
+import caseapp.core.ArgParser
+import org.broadinstitute.clio.transfer.model.{TransferWgsUbamV1Key, TransferWgsUbamV1QueryInput}
+import org.broadinstitute.clio.util.model.{DocumentStatus, Location}
+
+import scala.util.{Failure, Success, Try}
+
+object CustomArgParsers {
+  implicit def locationParser: ArgParser[Location] = {
+    ArgParser.instance[Location]("location") { location =>
```

Comment:
Same question as before: rather than having separate parsers for every `EnumEntry` type, could we have one general `ArgParser[T <: EnumEntry]`?

Reply:
We would still need to have an instance of check in order to know which enum to query. All we get from the arg parser is the string. I can try it though and you can let me know what you think.